opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchScrollRequest.java

```diff
@@ -165,11 +166,11 @@ public void writeTo(StreamOutput out) throws IOException {
    * Constructs OpenSearchScrollRequest from serialized representation.
    * @param in stream to read data from.
    * @param engine OpenSearchSqlEngine to get node-specific context.
-   * @throws IOException thrown if reading from input {@param in} fails.
+   * @throws IOException thrown if reading from input {@code in} fails.
    */
   public OpenSearchScrollRequest(StreamInput in, OpenSearchStorageEngine engine)
       throws IOException {
-    initialSearchRequest = new SearchRequest(in);
```

Comment:
Please add both a comment why this is necessary and a unit test that confirms this behaviour.

Serialization has one job -- ensure that `deserialize(serialize(obj))` is same as `obj`. Removing `initialSearchRequest` from `writeTo` breaks that assumption for `OpenSearchScrollRequest`.

If I'm not mistaken these were the reasons, right? 
1. `initialSearchRequest` is not needed after first search call and keeping it increases the size of the cursor we send out.
2. There is bug in core deserializing `SearchRequest`.

Reply:
`initialSearchRequest` is not a part which needs to be serialized. It is a `transient` field.
For reasons you listed: yes and yes.
I'll add more code comments.
Bug in deserialization for reference: https://github.com/opensearch-project/OpenSearch/issues/7829

Reply:
I understand why you are making `initialSearchRequest` transient and removing it from serialization, but from OO design, it is an error. Consider

```java
var r = new OpenSearchScrollRequest(...);
StreamOutput out = ...;
r.writeTo(out);
var r2 = new OpenSearchScrollRequest(out, ...);
r2.search(..., ...);
```

Assuming `r` and `out` are correctly initialized, `r.search(..., ...)` will succeed but `r2.search(.., ...)` will throw a null pointer exception.

Could you add a check in `search` to throw an `UnsupportedOperationException` if `isScroll()` is false and `intialSearchRequest` is null.

Reply:
Thank you, added in bd275665b.